apps/heureka/src/components/Services/useFetchServices.ts

```diff
@@ -15,21 +15,35 @@ type UseFetchServicesInput = {
 }
 
 export const useFetchServices = ({ filterSettings, pageSize = 10 }: UseFetchServicesInput) => {
+  const pagesRef = useRef<Page[]>()
   const [currentPage, setCurrentPage] = useState<number | undefined>(1)
   const { data, previousData, loading, error } = useGetServicesQuery({
     variables: {
       first: pageSize,
-      after: String(Math.abs(((currentPage || 1) - 1) * pageSize)),
+      after: pagesRef.current?.find((page) => page?.pageNumber === currentPage)?.after,
```

Comment:
whould it make sense to put the `pagesRef.current?.find((page) => page?.pageNumber === currentPage)?.after` so it can somehow be tested?

Reply:
Right now there are no tests for pagination because this will be changed to infinite scroll and there we can extract such things as small testable utils.